### PR TITLE
sensors: add baro calibration and cleanup

### PR DESCRIPTION
**Pull request by dagar**
The primary motivation here is exposing configurable barometer priorities and selecting external sensors by default, but it turned into a full baro calibration (offset) since it was easy to extend the existing pattern.

 - introduce baro cal (CAL_BAROx) mainly as a mechanism to adjust relative priority
 - sensor_baro.msg use SI (pressure in Pascals)
 - update all barometer drivers to publish directly and remove PX4Barometer helper
 - commander: add simple baro cal that sets baro offsets to align with
GPS altitude (if available)
 - create new sensors_status.msg to generalize sensor reporting

``` Console
nsh> listener sensor_baro

TOPIC: sensor_baro 2 instances

Instance 0:
 sensor_baro
    timestamp: 1144784841 (0.005478 seconds ago)
    timestamp_sample: 1144783899 (942 us before timestamp)
    device_id: 6780705 (Type: 0x67, I2C:4 (0x77))
    pressure: 101905.4688
    temperature: 40.6400
    error_count: 0

Instance 1:
 sensor_baro
    timestamp: 1144750525 (0.042671 seconds ago)
    timestamp_sample: 1144749606 (919 us before timestamp)
    device_id: 6780433 (Type: 0x67, I2C:2 (0x76))
    pressure: 101885.0547
    temperature: 40.6600
    error_count: 0




nsh> listener sensors_status_baro

TOPIC: sensors_status_baro
 sensors_status_baro
    timestamp: 1146569264 (0.020753 seconds ago)
    device_id_primary: 6780433 (Type: 0x67, I2C:2 (0x76))
    device_ids: [6780705, 6780433, 0, 0]
    inconsistency: [0.0988, -0.0976, nan, nan]
    healthy: [True, True, False, False]
    priority: [50, 75, 0, 0]
    enabled: [True, True, False, False]
    external: [False, True, False, False]



nsh> commander calibrate baro
nsh> INFO  [commander] [cal] calibration started: 2 baro
BARO 6780705 EN: 1, offset: 1323.242, 40.7 degC, Internal
BARO 6780433 EN: 1, offset: 1301.270, 40.7 degC, Ext
INFO  [commander] [cal] calibration done: baro

nsh> 
nsh> param show CAL_BARO*
Symbols: x = used, + = saved, * = unsaved
x + CAL_BARO0_ID [67,105] : 6780705
x + CAL_BARO0_OFF [68,106] : 1323.2422
x + CAL_BARO0_PRIO [69,107] : 50
x + CAL_BARO0_TEMP [70,108] : 40.6617
x + CAL_BARO1_ID [71,109] : 6780433
x + CAL_BARO1_OFF [72,110] : 1301.2695
x + CAL_BARO1_PRIO [73,111] : 75
x + CAL_BARO1_TEMP [74,112] : 40.6809
x   CAL_BARO2_ID [75,113] : 0
x   CAL_BARO3_ID [76,117] : 0

 712/1970 parameters used.
```

### CODE REVIEW

**Comment by dagar** [comment]
@ThomasDebrunner FYI this should fix the baro priorities

File: src/modules/sensors/vehicle_magnetometer/VehicleMagnetometer.cpp [hunk 1]
```diff
@@ -469,174 +469,218 @@ void VehicleMagnetometer::Run()
...
+				Vector3f current_mag{_last_data[i]};
 
-			float angle_error = AxisAnglef(Quatf(current_mag, primary_mag)).angle();
+				float angle_error = AxisAnglef(Quatf(current_mag, primary_mag)).angle();
 
-			// complementary filter to not fail/pass on single outliers
-			_mag_angle_diff[check_index] *= 0.95f;
-			_mag_angle_diff[check_index] += 0.05f * angle_error;
+				// complementary filter to not fail/pass on single outliers
+				_mag_angle_diff[check_index] *= 0.95f;
```
Comment (bresch) [inline src/modules/sensors/vehicle_magnetometer/VehicleMagnetometer.cpp:612]: we could also use the `AlphaFilter` class for this

**Review by bresch — DISMISSED** [review]

**Comment by sfuhrer** [comment]
> Pressure in simulation or something in the pressure handling got broken with these changes here I suppose. Tested with `make px4_sitl gazebo_tiltrotor` on current master. The reported pressure in sensor_baro is in mbar instead of Pa. Results in an invalid TAS, and that in turn in a correct "airspeed sensor failure detected". image

fixed by https://github.com/PX4/PX4-Autopilot/pull/19414

**Comment by dagar** [comment]
> Pressure in simulation or something in the pressure handling got broken with these changes here I suppose. Tested with `make px4_sitl gazebo_tiltrotor` on current master. The reported pressure in sensor_baro is in mbar instead of Pa. Results in an invalid TAS, and that in turn in a correct "airspeed sensor failure detected". image

For static pressure we had a mix of using millibar at the driver level and Pascals everywhere downstream. I tried to make this consistent everywhere, but apparently missed a few. Let me know if you have any suggestions for locking down the testing so this type of regression would have been caught in the first place. Sorry for the confusion.

**Pull request merged**